lib/AppInfo/Application.php

```diff
@@ -200,7 +200,7 @@ public function updateCSP() {
 
 		// Do not apply CSP rules on WebDAV/OCS
 		// Ideally this could be a middleware running after the controller execution before rendering the result to only do it on page response
-		if ($container->getServer()->getRequest()->getScriptName() !== '/index.php') {
+		if (!str_ends_with($container->getServer()->getRequest()->getScriptName(), '/index.php')) {
```

Comment:
Could you switch to something like substr_compare as we need to still support PHP 7.4 for the master branch and 7.3 when back porting this fix to stable release branches for Nextcloud 22/23?

Reply:
Now that was fast 😁